client-lite/src/util/do_json_parser.cpp

```diff
@@ -7,14 +7,25 @@
 std::chrono::seconds JsonParser::RefreshInterval = std::chrono::seconds(60);
 
 // Stores file path provided. Loads from the file later when a value is queried for.
-JsonParser::JsonParser(const std::string& jsonFilePath) :
+JsonParser::JsonParser(const std::string& jsonFilePath, bool alwaysCreateFile) :
     _jsonFilePath(jsonFilePath)
 {
+    if (!boost::filesystem::exists(_jsonFilePath) && alwaysCreateFile)
```

Comment:
>alwaysCreateFile [](start = 53, length = 16)

Check alwaysCreateFile first (cheaper) in the conditional. Same comment for the force check on line 28.

Reply:
It's okay but curious why you chose to split into separate if statements rather than combine with &&.

---
In reply to: [594671740](https://github.com/microsoft/do-client/pull/11#discussion_r594671740) [](ancestors = 594671740)